tests/base/src/test/java/org/wildfly/security/auth/realm/DistributedSecurityRealmTest.java

```diff
@@ -126,7 +148,66 @@ public void testBadEvidence() throws Exception {
         identity.dispose();
     }
 
+    @Test(expected = RealmUnavailableException.class)
+    public void testRealmUnavailable() throws Exception {
+        realm = new DistributedSecurityRealm(realm1, realm2, unavailableLdapRealm, realm3);
```

Comment:
Just to check, is the main difference between `unavailableRealm` and `unavailableLdapRealm` that one throws an exception when attempting to get an identity and the other throws an exception when attempting to check if an identity exists?

If so, may be be worth getting the body of this test into a separate method and creating two tests that each call this method, one can call it with `unavailableRealm` and the other can call it with `unavailableLdapRealm`.

A similar approach can be used for any appropriate tests below too.